src/projection.jl

```diff
@@ -1,24 +1,61 @@
 """
-    ProjectTo(x::T)
+    (p::ProjectTo{T})(dx)
 
-Returns a `ProjectTo{T,...}` functor able to project a differential `dx` onto the same tangent space as `x`.
-This functor encloses over what ever is needed to be able to be able to do that projection.
-For example, when projecting `dx=ZeroTangent()` on an array `P=Array{T, N}`, the size of `x`
-is not available from `P`, so it is stored in the functor.
+Projects the differential `dx` onto a specific cotangent space.
+This guaranees `p(dx)::T`, except for `dx::AbstractZero`.
 
-    (::ProjectTo{T})(dx)
+Usually `T` is the "outermost" part of the type, and it stores additional 
+properties in `backing(p)::NamedTuple`, such as projectors for each constituent
+field, and a projector `p.element` for the element type of an array of numbers.
 
-Projects the differential `dx` on the onto the tangent space used to create the `ProjectTo`.
+When called on `dx::Thunk`, the projection is inserted into the thunk.
 """
 struct ProjectTo{P,D<:NamedTuple}
     info::D
 end
 ProjectTo{P}(info::D) where {P,D<:NamedTuple} = ProjectTo{P,D}(info)
 ProjectTo{P}(; kwargs...) where {P} = ProjectTo{P}(NamedTuple(kwargs))
 
-backing(project::ProjectTo) = getfield(project, :info)
+"""
+    ProjectTo(x)
+
+Returns a `ProjectTo{T}` functor which projects a differential `dx` onto the
+relevant cotangent space for `x`.
+
+At present this undersands only `x::AbstractArray`, `x::Number` and `x::Ref`. 
+It should not be called on arguments of an `rrule` method which accepts other types.
+
+# Examples
+```jldoctest
```

Comment:
Doctest seems to fail completely: https://github.com/JuliaDiff/ChainRulesCore.jl/pull/391/checks?check_run_id=3022855366

Reply:
Do we need to add a `using LinearAlgebra` here
https://github.com/JuliaDiff/ChainRulesCore.jl/blob/master/docs/make.jl#L13 
?

Reply:
Great, that seems to fix it.